Scripts/Runtime/Core/AbstractAnvilMonoBehaviour.cs

```diff
@@ -50,6 +50,11 @@ protected virtual void Awake()
         /// Will Destroy the attached <see cref="GameObject"/> if it has not been destroyed yet.
         /// </summary>
         public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        private void Dispose(bool shouldDestroyGameObject)
```

Comment:
Might make sense to make this public some day if you only want to dispose the specific behaviour but we should wait until the need actually comes up.

We'd need a `RemoveComponent` path anyway.

Reply:
Yeah that's fair. Interesting that in all these years that's never been a pattern we've done.

Reply:
Yea, the whole dispose pattern is a bit of an opinionated choice.

On one hand you could say that we should have `Dispose` at all and developers should just use the engine commands `Object.Destroy` and `MonoBehaviour.RemoveComponent`.

On the other hand, it's nice to implement the `IDisposable` pattern and, personally, I prefer to call `Dispose` vs using Unity's various ways of destroying objects.

Reply:
Agreed!